host.py

```diff
@@ -294,23 +295,42 @@ def copy_to(self, src_file: str, dst_file: str) -> None:
     def need_sudo(self) -> None:
         self.sudo_needed = True
 
-    def run(self, cmd: str, log_level: int = logging.DEBUG, env: dict[str, str] = os.environ.copy()) -> Result:
+    def run(
+        self,
+        cmd: str,
+        log_level: int = logging.DEBUG,
+        *,
+        env: Optional[Mapping[str, Optional[str]]] = None,
```

Comment:
can we set `env: Mapping[str, Optional[str]] = some_defaults()` and simplify the code?

Reply:
I don't understand. How do you suggest to simplify the code (or how does `some_defaults()` allow for simplification)?

Reply:
You pass in `None` to trigger the `if env` in the code. Take the body of the `if` and put that in a function and assign it like this. `env: Mapping[str, Optional[str]] = the_extracted_function_name()`.